core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -899,6 +926,28 @@ private fun Preview.Builder.updateCameraStateWithCaptureResults(
                 result: TotalCaptureResult
             ) {
                 super.onCaptureCompleted(session, request, result)
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+                    val boostState = result.get(CaptureResult.CONTROL_LOW_LIGHT_BOOST_STATE)
+                    when (boostState) {
+                        CameraMetadata.CONTROL_LOW_LIGHT_BOOST_STATE_ACTIVE -> {
+                            currentCameraState.update { old ->
+                                old.copy(
+                                    lowLightBoostState = LowLightBoostState.ACTIVE
+                                )
+                            }
+                        }
+
+                        CameraMetadata.CONTROL_LOW_LIGHT_BOOST_STATE_INACTIVE -> {
+                            currentCameraState.update { old ->
+                                old.copy(
+                                    lowLightBoostState = LowLightBoostState.INACTIVE
+                                )
+                            }
+                        }
+                    }
+                }
```

Comment:
Similar to above, you should check if the LLB state has actually changed before calling `copy`. Something like this:
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
    val nativeBoostState = result.get(CaptureResult.CONTROL_LOW_LIGHT_BOOST_STATE)
    val boostState = when (nativeBoostState) {
        CameraMetadata.CONTROL_LOW_LIGHT_BOOST_STATE_ACTIVE -> LowLightBoostState.ACTIVE
        else -> LowLightBoostState.INACTIVE
    }
    currentCameraState.update { old ->
        if (old.lowLightBoostState != boostState) {
            old.copy(lowLightBoostState = boostState)
        } else {
            old
        }
    }
}
```